pkg/apis/sources/v1beta1/ping_conversion.go

```diff
@@ -106,94 +81,18 @@ func (sink *PingSource) ConvertFrom(ctx context.Context, obj apis.Convertible) e
 			SourceStatus: source.Status.SourceStatus,
 		}
 
-		// deep copy annotations to avoid mutation on source.ObjectMeta.Annotations
-		annotations := make(map[string]string)
-		for key, value := range source.GetAnnotations() {
-			annotations[key] = value
-		}
-
-		if isV1Beta1AnnotationConsistentWithV1Beta2Spec(source) {
-			// errors already handled in isV1Beta1AnnotationConsistentWithV1Beta2Spec
-			v1beta1Spec := annotations[V1B1SpecAnnotationKey]
-			_ = json.Unmarshal([]byte(v1beta1Spec), &sink.Spec)
+		sink.Spec = PingSourceSpec{
+			SourceSpec: source.Spec.SourceSpec,
+			Schedule:   source.Spec.Schedule,
+			Timezone:   source.Spec.Timezone,
 		}
 
-		// marshal and store v1beta2.PingSource.Spec into V1B2SpecAnnotationKey
-		// this is to help if we need to convert back to v1beta2.PingSource
-		v1beta2Configuration, err := json.Marshal(source.Spec)
-		if err != nil {
-			return fmt.Errorf("error marshalling source.Spec: %v, err: %v", source.Spec, err)
+		if source.Spec.ContentType == cloudevents.ApplicationJSON {
+			sink.Spec.JsonData = source.Spec.Data
```

Comment:
@lionelvillard do we want to populate the payload when contentType is not json?  I prefer no because it contradicts with the name `JsonData`.  what about spec.dataBase64?

Reply:
I don't think it matters. dataBase64 is v1beta2 only so it is expected to not be in v1beta1. 

What about checking is source.Spec.Data is valid JSON. If yes, set JsonData, if not, what about raising an error?

Reply:
> What about checking is source.Spec.Data is valid JSON. If yes, set JsonData, if not, what about raising an error?
It is already checked in `pingsource_validation` so it's always valid, see https://github.com/knative/eventing/blob/main/pkg/apis/sources/v1beta2/ping_validation.go#L87-L91

Reply:
I mean instead of relying on `ContentType`, always check if `Data` is valid JSON and if yes set `JsonData`. `pingsource_validation` only handles the case when `ContentType` is set to a specific value.

BTW, great job as simplifying this code. Looks great!

Reply:
I see, but `ContentType=text/plain, Data={"msg": "hello"}` is also a v1beta2 feature. 

Also if we do set JsonData, from a v1beta1 client's perspective, the expected cloudevent datacontenttype on the wire would be `application/json` instead of `text/plain`. The question is how do we present something like `ContentType=text/plain, Data={"msg": "hello"}`  to v1beta1 clients, do we throw error or return an abridged spec.?

Reply:
I would throw an error to avoid surprises.

Reply:
In this case, we might want to change [kn](https://github.com/knative/client) to use the v1beta2 API for PingSource as well (right now it's using v1alpha2).

Reply:
@lionelvillard I took a look at this, https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-request-and-response 

it says 
> Failing conversion can disrupt read and write access to the custom resources, including the ability to update or delete the resources. Conversion failures should be avoided whenever possible

I'm thinking maybe we can always succeed the conversion, for incompatible cases, just leave JsonData as empty, something like this:

```shell
$ kubectl get pingsources.v1beta1.sources.knative.dev test-ping-source-xml -n events -o=jsonpath='{.spec}' | jq
{
  "schedule": "*/1 * * * *",
  "sink": {
    "ref": {
      "apiVersion": "v1",
      "kind": "Service",
      "name": "event-display"
    }
  }
}

$ kubectl get pingsources.v1beta2.sources.knative.dev test-ping-source-xml -n events -o=jsonpath='{.spec}' | jq
{
  "contentType": "application/xml",
  "data": "<ok>hello</ok>",
  "schedule": "*/1 * * * *",
  "sink": {
    "ref": {
      "apiVersion": "v1",
      "kind": "Service",
      "name": "event-display"
    }
  }
}
```

I believe this has better user experience as it at least does not break the existing read experience, for writing v1beta2 PingSource via v1beta1 API, the end result will be ContentType="application/json". 

WDYT?